extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcBuildTimeConfig.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.oidc.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * Build time configuration for OIDC.
+ */
+@ConfigRoot
+public class OidcBuildTimeConfig {
```

Comment:
@dmlloyd It looks good in that an `ApplicationType` and 'enabled' properties should only be checked at the build time. Can you please explain how splitting it away from `OidcConfig` helps ?

Reply:
I see, that means it can not be modified at the runtime. I guess it is OK then

Reply:
By the way, would extending work ? `OidcConfig` extending `OidcBuildTimeConfig` ?

Reply:
Extending does not work unfortunately.  The build time config and run time config have to be separate classes.